src/Adapters/Auth/AuthAdapter.js

```diff
@@ -1,20 +1,61 @@
 /*eslint no-unused-vars: "off"*/
 export class AuthAdapter {
-  /*
+  constructor() {
+    /**
+     * If set to true the signup/login
+     * will fail if user do not provide
+     * authData associated to this Auth Adapter
+     */
+    this.required = false;
+
+    /**
+     * Set to true if you want to force to validateAuthData
+     * even if authData do not change
+     */
+    this.alwaysValidate = false;
+
+    /**
+     * Usage policy
+     * default: can be combined with ONE additional auth provider if additional configured on user
+     * additional: could be only used with a default policy auth provider
+     * solo: Will ignore ALL additional providers if additional configured on user
+     */
+    this.policy = 'default';
```

Comment:
I didn't get it well. We can't have multilpe adapters anymore, let's say for google, facebook, twitter?

Reply:
Yes we need to improve this doc part. Policy allow parse server to know how to handle 2FA. default is the policy that always existed in parse server, a user can login/signup directly with just `facebook` or `twitter` etc... adapters with default policy always have an ID to identify a user

Then we will have the MFA adapter with `additional` policy, additional policy tell to parse server that if this kind of auth is configured on the user, it need to be provided as a second factor authentication method like `facebook + MFA`, `twitter + MFA`, `username + MFA`, `facebook + SMS OTP`; Auth Adapters with additional policy DO NOT HAVE ids , since there will be used as a validation auth system (OTP, MFA, etc...) and are not intended to be used as an identifier auth system.

Finally the policy `solo` tell to the parse server that the auth adapter (like webauthn) DO NOT NEED additionnal validation if an additional auth adapter (like MFA) was configured on the User. 

This way we have:
`default`: Intended to signup/login/identify the user and can be combined with an additional auth system if the user has one configured
`additional`: Intended to be used as a 2FA to validate an initial login via a `default` adapter
`solo`: Can be used to signup/login/identify the user and CAN'T be combined with an additional auth system

I am quite open if you have an idea to clarify as much as possible these behaviors :)

Note the system that i designed only support 2FA (not 3FA,4FA since it's useless), so if a user has for example: username/password (considered as default), he has SMS OTP (additional) and also MFA (additional), parse server will require the user to provide ONE OF the additional auth system, in this use case `username/password/OTP` OR `username/password/MFA` the client currently has the choice about which configured additional auth system he wants to use.

Reply:
Ok. Now I understand. Maybe this explanation should go in the code or docs :)